.drone.yml

```diff
@@ -1,297 +1,391 @@
-      event: [push, tag, deployment]
-
-  deploy-to-cs-dev-from-build-number:
-    group: deploy-to-dev
+  - name: deploy to cs-qa
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=cs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+      - ENVIRONMENT: cs-qa
+      - KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_cs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - CS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
 
-  deploy-to-wcs-dev-from-build-number:
-    group: deploy-to-dev
+  - name: deploy to wcs-qa
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=wcs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+      - ENVIRONMENT: wcs-qa
+      - KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_wcs
+      - hocs_info_service_wcs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - WCS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
 
-  deployment:
+  - name: deployment to not-prod
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=${DRONE_DEPLOY_TO}
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_dev_cs
-      - hocs_info_service_demo_cs
-      - hocs_info_service_dev_wcs
-      - hocs_info_service_demo_wcs
+      ENVIRONMENT: ${DRONE_DEPLOY_TO}
+      KUBE_TOKEN:
+        from_secret: hocs_info_service_${DRONE_DEPLOY_TO/-/_}
     commands:
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
     when:
-      event: deployment
-      environment: [cs-dev, cs-demo, wcs-dev, wcs-demo]
+      target:
+        exclude:
+          - release
+          - restart
+          - "*-prod"
+      event:
+        - promote
 
-  deploy-to-qa-cs:
+  - name: deploy to prod
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=cs-qa
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_qa_cs
+      ENVIRONMENT: ${DRONE_DEPLOY_TO}
+      KUBE_TOKEN:
+        - hocs_info_service_${DRONE_DEPLOY_TO/-/_}
```

Comment:
```suggestion
      KUBE_TOKEN: hocs_info_service_${DRONE_DEPLOY_TO/-/_}
```

Reply:
Needed to come secrets - this has been changed 👍🏼